cmd/fleet/serve.go

```diff
@@ -871,7 +871,24 @@ the way that the Fleet server works.
 				}
 			}
 
-			rootMux.Handle("/api/", apiHandler)
+			// We must wrap the Handler here to set special per-endpoint Write
+			// timeouts, so that we have access to the raw http.ResponseWriter.
+			// Otherwise, the handler is wrapped by the promhttp response delegator,
+			// which does not support the Unwrap call needed to work with
+			// ResponseController.
+			//
+			// See https://pkg.go.dev/net/http#NewResponseController which explains
+			// the Unwrap method that the prometheus wrapper of http.ResponseWriter
+			// does not implement.
+			rootMux.HandleFunc("/api/", func(rw http.ResponseWriter, req *http.Request) {
+				if req.Method == http.MethodPost && strings.HasSuffix(req.URL.Path, "/fleet/scripts/run/sync") {
+					rc := http.NewResponseController(rw)
+					if err := rc.SetWriteDeadline(time.Now().Add((5 * time.Minute) + (1 * time.Second))); err != nil {
+						level.Error(logger).Log("msg", "http middleware failed to override endpoint write timeout", "err", err)
+					}
+				}
+				apiHandler.ServeHTTP(rw, req)
+			})
```

Comment:
This is ugly, but the issue is in the `promhttp` third-party package, they wrap the raw `http.ResponseWriter` but don't provide the `Unwrap` method required to unwrap it back to the original value, and this is required for `ResponseController` to work properly. 

In the meantime and pressed by time I went with this ugly approach, but I'm sure the prometheus folks would be open to integrate the change as this is the right way to do it since Go 1.20, and then we could have a cleaner approach and do this timeout extension where the route is defined, e.g. with a `ue.WithHTTPMiddleware(...).POST("/...")` (which I attempted at first before noticing the prometheus thing).